plugin/src/main/java/io/jenkins/plugins/coverage/metrics/steps/CoverageApiUtil.java

```diff
@@ -0,0 +1,90 @@
+package io.jenkins.plugins.coverage.metrics.steps;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import edu.hm.hafner.coverage.FileNode;
+import edu.hm.hafner.coverage.Node;
+
+import io.jenkins.plugins.coverage.metrics.model.FileWithModifiedLines;
+import io.jenkins.plugins.coverage.metrics.model.LineCoverageType;
+import io.jenkins.plugins.coverage.metrics.model.ModifiedLinesBlock;
+
+/**
+ * Server side model that provides data for the details of line coverage results of modified lines.
+ */
+public class CoverageApiUtil {
```

Comment:
This class and methods should be actually part of the API class. 

Then the constructor of `LineCoverageApi` works on a `Node`. You can make the method package private to test it there.

Reply:
We have been on the fence about this. The logic is now part of the API class again.